osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -154,9 +154,18 @@ public virtual WorkingBeatmap CreateNewDifficulty(BeatmapSetInfo targetBeatmapSe
                 DifficultyName = NamingUtils.GetNextBestName(targetBeatmapSet.Beatmaps.Select(b => b.DifficultyName), "New Difficulty")
             };
             var newBeatmap = new Beatmap { BeatmapInfo = newBeatmapInfo };
+
             foreach (var timingPoint in referenceWorkingBeatmap.Beatmap.ControlPointInfo.TimingPoints)
                 newBeatmap.ControlPointInfo.Add(timingPoint.Time, timingPoint.DeepClone());
 
+            foreach (var effectPoint in referenceWorkingBeatmap.Beatmap.ControlPointInfo.EffectPoints)
+            {
+                if (!rulesetInfo.Equals(referenceWorkingBeatmap.BeatmapInfo.Ruleset))
+                    effectPoint.ScrollSpeedBindable.SetDefault();
```

Comment:
Shouldn't this be operating on already-cloned effect points, otherwise it's affecting the reference beatmap?

Reply:
Ouch good catch.